### ISSUE TITLE
Keyset cursor could contain non-base64 symbols

### ISSUE THREAD

**Issue by dolfinus** [ash-project/ash#337]
**Describe the bug**
If resource can be identified not only by id, but some other field or fields, this field is used while generating value of cursors for Keyset paginagion.

In some cases generated value is not a valid base64 string.

**To Reproduce**
```elixir
defmodule AshGraphql.Test.RelayTag do
  use Ash.Resource,
    data_layer: Ash.DataLayer.Ets

  actions do
    defaults([:create, :update, :destroy, :read])

    read :read_paginated do
      pagination(required?: true, keyset?: true, countable: true)
    end
  end

  attributes do
    uuid_primary_key(:id)

    attribute(:name, :string)
  end

  identities do
    identity(:name, [:name], pre_check_with: AshGraphql.Test.Api)
  end
end

defmodule AshGraphql.Test.Registry do
  @moduledoc false
  use Ash.Registry

  entries do
    entry(AshGraphql.Test.RelayTag)
  end
end


defmodule AshGraphql.Test.Api do
  @moduledoc false

  use Ash.Api

  # graphql do
  #  authorize? true
  #  debug? false
  #  show_raised_errors? true
  # end

  resources do
    registry(AshGraphql.Test.Registry)
  end
end


letters = ["a", "b", "c", "d", "e", "f", "g"]

for name <- letters do
  tag =
    AshGraphql.Test.RelayTag
    |> Ash.Changeset.for_create(
      :create,
      [name: name]
    )
    |> AshGraphql.Test.Api.create!()
end

{:ok, page1} = AshGraphql.Test.Api.read(%Ash.Query{resource: AshGraphql.Test.RelayTag, sort: [name: :asc], select: [:name]}, action: :read_paginated, page: [limit: 4])

first_record_page1 = page1.results |> Enum.at(0)

# %AshGraphql.Test.RelayTag{
#   __meta__: #Ecto.Schema.Metadata<:built, "">,
#   __metadata__: %{keyset: "g2wAAAABbQAAAAFhag%3D%3D", selected: [:name]},
#   __order__: nil,
#   aggregates: %{},
#   calculations: %{},
#   id: "92004b93-379b-43dc-a62f-1474f5571a59",
#   name: "a"
# }

first_record_page1.__metadata__.keyset
# "g2wAAAABbQAAAAFhag%3D%3D"
```

`%3D` is urlencoded `=` symbol.

If resource does not contain `identity(:name)`, other cursor is returned which looks fine:
```elixir
defmodule AshGraphql.Test.RelayTag do
  use Ash.Resource,
    data_layer: Ash.DataLayer.Ets

  actions do
    defaults([:create, :update, :destroy, :read])

    read :read_paginated do
      pagination(required?: true, keyset?: true, countable: true)
    end
  end

  attributes do
    uuid_primary_key(:id)

    attribute(:name, :string)
  end
end

...

{:ok, page1} = AshGraphql.Test.Api.read(%Ash.Query{resource: AshGraphql.Test.RelayTag, sort: [name: :asc], select: [:name]}, action: :read_paginated, page: [limit: 4])

first_record_page1 = page1.results |> Enum.at(0)

# %AshGraphql.Test.RelayTag{
#   __meta__: #Ecto.Schema.Metadata<:built, "">,
#   __metadata__: %{
#     keyset: "g2wAAAACbQAAACQyNjcyZjVhZC1iYTBiLTQwZTAtOTM1Ni0wMDQ4YWJhNzU4NzltAAAAAWFq",
#     selected: [:name]
#   },
#   __order__: nil,
#   aggregates: %{},
#   calculations: %{},
#   id: "2672f5ad-ba0b-40e0-9356-0048aba75879",
#   name: "a"
# }

first_record_page1.__metadata__.keyset
# "g2wAAAACbQAAACQyNjcyZjVhZC1iYTBiLTQwZTAtOTM1Ni0wMDQ4YWJhNzU4NzltAAAAAWFq"
```

Also if identity is present, but sorting is performed over other field, the resulting cursor is also fine:
```elixir
{:ok, page1} = AshGraphql.Test.Api.read(%Ash.Query{resource: AshGraphql.Test.RelayTag, select: [:name]}, action: :read_paginated, page: [limit: 4])

first_record_page1 = page1.results |> Enum.at(0)

# %AshGraphql.Test.RelayTag{
#   __meta__: #Ecto.Schema.Metadata<:built, "">,
#   __metadata__: %{
#     keyset: "g2wAAAABbQAAACQwZTk5NzdjMS05NTUzLTRkMGYtOTNjNi1jYmMwZTk1YmRlZDJq",
#     selected: [:name]
#   },
#   __order__: nil,
#   aggregates: %{},
#   calculations: %{},
#   id: "0e9977c1-9553-4d0f-93c6-cbc0e95bded2",
#   name: "b"
# }

first_record_page1.__metadata__.keyset
# "g2wAAAABbQAAACQwZTk5NzdjMS05NTUzLTRkMGYtOTNjNi1jYmMwZTk1YmRlZDJq"
```

**Expected behavior**
Cursor values should be a valid base64 string, not containing urlencoded parts.

**Runtime**
- Elixir 1.13.2 (compiled with Erlang/OTP 24)
- Erlang/OTP 24 [erts-12.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit]
- Linux 5.18.0-1-MANJARO
- 1.52.0-rc.9

**Additional context**

Maybe keyset pagination should always return cursor based on record primary key instead of identity value? #255 is exactly about this

**Comment by zachdaniel** [comment]
Yep! #255 ought to involve fixing this as well.

**Comment by zachdaniel** [comment]
The actual fix here was just a simple removal of the call to `    |> URI.encode_www_form()`.

If something else needs to do url encode it that is fine, as we will first attempt to `decode_www_form()` on the input, just in case as it seems like a very common use case to pass in the value via a query parameter.

**Comment by zachdaniel** [comment]
Fixed in 393dd7326f567f2aa3cdae06700cdee66f846c5a
